tools/test/examples/update.py

```diff
@@ -72,113 +94,167 @@ def upgrade_single_example(example, tag, directory):
+    user - GitHub user name
+    ref - SHA corresponding to the tag
     
     """
-    print("Updating example '%s'" % example['name'])
+    ret = False
+    print("\nUpdating example '%s'" % example['name'])
     cwd = os.getcwd()
-    
-    # Setup and run the import command
-    clone_cmd = ['git', 'clone', example['github']]
-    return_code = run_cmd(clone_cmd)
-    
-    if return_code:
+
+    full_repo_name = 'ARMmbed/'+ example['name']
+    fork = "https://github.com/" + user + '/' + example['name'] 
+
+    # Check access to mbed-os repo
+    try:
+        repo = github.get_repo(full_repo_name, False)
+
+    except:
+        print("\t\t!! Repo does not exist - skipping\n")
         return False
+
+
+    # Clone the forked example repo
+    clone_cmd = ['git', 'clone', fork]
+    return_code = run_cmd(clone_cmd)
     
-    # Find all examples
-    example_directories = find_all_examples(example['name'])
-    
-    os.chdir(example['name'])
+    if not return_code:
     
-    # Setup and run the update command
-    import_cmd = ['mbed', 'update']
-    return_code = run_cmd(import_cmd)
-    if return_code:
-        os.chdir(cwd)
-        return False
-    
-    for example_directory in example_directories:
-        if not upgrade_single_example(example, tag, os.path.relpath(example_directory, example['name'])):
-            os.chdir(cwd)
-            return False
+        # Find all examples
+        example_directories = find_all_examples(example['name'])
+        
+        os.chdir(example['name'])
         
-    # Setup the default commit message
-    commit_message = 'Updating mbed-os to {{' + tag +'}}'
+        # checkout and synchronise the release-candidate branch
+        prepare_fork(example['github'])
+        
+        for example_directory in example_directories:
+            if not upgrade_single_example(example, tag, os.path.relpath(example_directory, example['name']), ref):
+                os.chdir(cwd)
+                return False
+
+        # Setup the default commit message
+        commit_message = 'Updating mbed-os to ' + tag 
+
+        # Setup and run the commit command
+        commit_cmd = ['git', 'commit', '-m', commit_message]
+        return_code = run_cmd(commit_cmd)
+        if not return_code:
+
+            # Setup and run the push command
+            push_cmd = ['git', 'push', 'origin']
+            return_code = run_cmd(push_cmd)
             
-    # Setup and run the commit command
-    commit_cmd = ['git', 'commit', '-m', commit_message]
-    return_code = run_cmd(commit_cmd)
-    if return_code:
-        if return_code == 1:
-            print("[WARNING] 'git commit' exited with a return code of 1. " + \
-            "This usually inidicates that no update was made. Still " + \
-            "attempting to create a tag.")
+            if not return_code:
+                body = "Please test/merge this PR and then tag Master with " + tag
```

Comment:
Until all the example owners have automatic testing of their examples for each PR, then they need the prompt!